openerp/models.py

```diff
@@ -2160,6 +2160,14 @@ def read_group(self, cr, uid, domain, fields, groupby, offset=0, limit=None, con
             for d in fetched_data:
                 d.update(data_dict[d['id']])
 
+        selectionfields = [gb['field'] for gb in annotated_groupbys if gb['type'] == 'selection']
+        if selectionfields:
+            selections_values = {}
+            for selection in selectionfields:
+                selections_values[selection] = dict(self._fields[selection].selection)
+            for d in fetched_data:
+                d.update(dict((selection, selections_values[selection][d[selection]]) for selection in selectionfields))
```

Comment:
You don't need to build a dict here, `update` can take a sequence of (key, value) pairs.